### PR TITLE
Convert Donate Page to TypeScript

### LINKED ISSUE #221: Convert Donate Page to TypeScript
https://github.com/ShelterTechSF/sheltertech.org/blob/main/src/pages/donate/index.js

### PR DESCRIPTION
**Pull request by candywang**
Closes #221.

### CODE REVIEW

**Comment by candywang** [comment]
I removed the module declaration for react-helmet. Once #227 is merged, this branch can pull and merge the changes to fix the TypeScript error.

**Review by richardxia — APPROVED** [review]
LGTM once you pull in the latest `main` branch with the react-helmet types pulled in!

**Review by davidagustin — APPROVED** [review]
Lgtm

**Pull request merged**
